File: examples/simple_lambda_test.py
"""Simple lambda function tests"""

def test_basic_lambdas():
    # Single parameter lambda
    square = lambda x: x * x
    result1 = square(5)
    
    # Two parameter lambda
    add = lambda x, y: x + y
    result2 = add(3, 4)
    
    # No parameter lambda
    get_value = lambda: 42
    result3 = get_value()
    
    return result1 + result2 + result3

File: examples/simple_lambda_test.rs
#[doc = " Depyler: verified panic-free"]
#[doc = " Depyler: proven to terminate"]
pub fn test_basic_lambdas() -> i32 {
    let square = |x| x * x;
    let result1 = square(5);
    let add = |x, y| x + y;
    let result2 = add(3, 4);
    let get_value = || 42;
    let result3 = get_value();
    result1 + result2 + result3
}
